rmw_fastrtps_shared_cpp/src/qos.cpp

```diff
@@ -157,14 +157,13 @@ get_datawriter_qos(
 }
 
 bool
-is_valid_qos(
-  const rmw_qos_profile_t & qos_policies)
+is_valid_qos(const rmw_qos_profile_t & /*qos_policies*/)
 {
-  if (qos_policies.liveliness == RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE ||
-    qos_policies.liveliness == RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC)
-  {
-    RMW_SET_ERROR_MSG("Manual liveliness unsupported for fastrtps");
-    return false;
-  }
+  // if (qos_policies.liveliness == RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE ||
+  //   qos_policies.liveliness == RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC)
+  // {
+  //   RMW_SET_ERROR_MSG("Manual liveliness unsupported for fastrtps");
```

Comment:
remove instead of comment?

Reply:
This PR also needs to go in for FastRTPS 1.8.0, not just https://github.com/ros2/rmw_fastrtps/pull/272, right? In which case we'll need to uncomment it again.

This code should definitely be gone for 1.8.1 though (assuming liveliness becomes available by then).

Reply:
Robert C. Martin's Clean Code Tip of the Week #7: Clean up Old Commented Out Code
http://www.informit.com/articles/article.aspx?p=1334908

> It makes me crazy to see stretches of code that are commented out. Who knows how old it is? Who knows whether or not it’s meaningful? Yet no one will delete it because everyone assumes someone else needs it or has plans for it.
> 
> That code sits there and rots, getting less and less relevant with every passing day. It calls functions that no longer exist. It uses variables whose names have changed. It follows conventions that are long obsolete. It pollutes the modules that contain it and distracts the people who try to read it. Commented-out code is an abomination.
> 
> When you see commented-out code, delete it! Don’t worry; the source code control system still remembers it. If anyone really needs it, he or she can go back and check out a previous version. Don’t suffer commented-out code to survive.

Reply:
If this PR is intending to enable automatic liveliness, and not manual liveliness, then uncomment it. If this PR is intending to enable all 3 liveliness policies, then delete the block. I don't think  we need a middle ground.